sklearn/_fast_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -0,0 +1,194 @@
+                                       n_estimators=n_estimators,
+                                       n_iter_no_change=n_iter_no_change,
+                                       random_state=0)
+    gb.fit(X, y)
+
+    if n_iter_no_change is not None:
+        assert n_iter_no_change <= gb.n_estimators_ < n_estimators
+    else:
+        assert gb.n_estimators_ == n_estimators
+
+
+@pytest.mark.parametrize('data', (
+    make_classification(random_state=0),
+    make_classification(n_classes=3, n_clusters_per_class=1, random_state=0)
+))
+@pytest.mark.parametrize(
+    'scoring, validation_fraction, n_iter_no_change, tol', [
+        ('accuracy', .1, 5, 1e-7),  # use scorer
+        ('accuracy', None, 5, 1e-1),  # use scorer on training data
+        (None, .1, 5, 1e-7),  # same with default scorerscor
+        (None, None, 5, 1e-1),
+        ('loss', .1, 5, 1e-7),  # use loss
+        ('loss', None, 5, 1e-1),  # use loss on training data
+        (None, None, None, None),  # no early stopping
+        ])
+def test_early_stopping_classification(data, scoring, validation_fraction,
+                                       n_iter_no_change, tol):
+
+    n_estimators = 50
+
+    X, y = data
+
+    gb = FastGradientBoostingClassifier(
+        verbose=1,  # just for coverage
+        scoring=scoring,
+        tol=tol,
+        validation_fraction=validation_fraction,
+        n_estimators=n_estimators,
+        n_iter_no_change=n_iter_no_change,
+        random_state=0)
+    gb.fit(X, y)
+
+    if n_iter_no_change is not None:
+        assert n_iter_no_change <= gb.n_estimators_ < n_estimators
+    else:
+        assert gb.n_estimators_ == n_estimators
+
+
+def test_should_stop():
+
+    def should_stop(scores, n_iter_no_change, tol):
+        gbdt = FastGradientBoostingClassifier(
+            n_iter_no_change=n_iter_no_change,
+            tol=tol)
+        return gbdt._should_stop(scores)
+
+    # not enough iterations
+    assert not should_stop([], n_iter_no_change=1, tol=0.001)
+
+    assert not should_stop([1, 1, 1], n_iter_no_change=5, tol=0.001)
+    assert not should_stop([1] * 5, n_iter_no_change=5, tol=0.001)
+
+    # still making significant progress up to tol
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.001)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.999)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5,
+                           tol=5 - 1e-5)
+
+    # no significant progress according to tol
+    assert should_stop([1] * 6, n_iter_no_change=5, tol=0.)
+    assert should_stop([1] * 6, n_iter_no_change=5, tol=0.001)
+    assert should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=5)
+
+
+@pytest.mark.parametrize('Estimator', (
+    FastGradientBoostingRegressor(),
+    FastGradientBoostingClassifier(),
+    ))
+def test_estimator_checks(Estimator):
```

Comment:
According to [travisci](https://travis-ci.org/scikit-learn/scikit-learn/jobs/486498378#L3175) these tests are the slowest in the test suite:

```bash
24.73s call     _fast_gradient_boosting/tests/test_gradient_boosting.py::test_estimator_checks[Estimator1]
14.32s call     _fast_gradient_boosting/tests/test_gradient_boosting.py::test_estimator_checks[Estimator0]
```

Not sure if there is anything to be done about this.

Reply:
Yes these should be deleted ultimately since they are already run in `test_common`. I'm just keeping them here temporarily so that we don't have to run the whole test suite to check whether they'll pass.